readthedocs/projects/admin.py

```diff
@@ -177,6 +178,35 @@ def delete_selected_and_artifacts(self, request, queryset):
                 broadcast(type='app', task=remove_dir, args=[project.doc_path])
         return delete_selected(self, request, queryset)
 
+    def reindex_active_versions(self, request, queryset):
+        """Reindex all active versions of selected projects"""
+        for project in queryset:
+            versions_qs = Version.objects.filter(project__slug=project.slug)
+            if not versions_qs.exists():
+                error_msg = 'No project with slug {}'.format(project.slug)
+                self.message_user(request, error_msg, level=messages.ERROR)
+
+            active_versions = versions_qs.filter(active=True)
+
+            for version in active_versions:
+                try:
+                    commit = version.project.vcs_repo(version.slug).commit
+                except:  # noqa
+                    # An exception can be thrown here in production, but it's not
+                    # documented what the exception here is.
+                    commit = None
+
+                try:
+                    update_search(version.pk, commit,
+                                  delete_non_commit_files=False)
+                    success_msg = 'Reindexing triggered for {}'.format(version)
```

Comment:
I am thinking to move this try/except block to a separate function because it will be used for Version admin also.
Any advice on this?